functions/Add-DbaAgDatabase.ps1

```diff
@@ -137,18 +137,28 @@ function Add-DbaAgDatabase {
             $allbackups = @{
             }
 
-            $Primary = $db.Parent
+            $primary = $db.Parent
             # check primary, should be run against primary
-            $ag = Get-DbaAvailabilityGroup -SqlInstance $db.Parent -AvailabilityGroup $AvailabilityGroup
+            $ag = Get-DbaAvailabilityGroup -SqlInstance $primary -AvailabilityGroup $AvailabilityGroup
 
             if (-not $ag.Parent) {
-                Stop-Function -Message "Availability Group $AvailabilityGroup not found on $($db.Parent.Name)" -Continue
+                Stop-Function -Message "Availability Group $AvailabilityGroup not found on $($primary.Name)" -Continue
             }
 
             if ($ag.AvailabilityDatabases.Name -contains $db.Name) {
                 Stop-Function -Message "$($db.Name) is already joined to $($ag.Name)" -Continue
             }
 
+            if ($SeedingMode -eq "Automatic" -and $primary.VersionMajor -lt 13) {
+                Stop-Function -Message "Automatic seeding mode only supported in SQL Server 2016 and above" -Continue
+            }
+
+            # disabled, we first have to discuss the need for this test and the consequences of this test
+            # was not active in previous version due to wrong placement
+            #            if (-not $UseLastBackup -and -not $SharedPath -and $SeedingMode -ne 'Automatic') {
+            #                Stop-Function -Message "You must specify a SharedPath when adding databases to a manually seeded availability group" -Continue
+            #            }
+
```

Comment:
Why would we not want to verify that either `UseLastBackup` or `SharedPath` is provided when seeding mode defaults to manual?

Reply:
I'll start a discussion about that as soon as this pull request is merged. 
Short answer is: Because then the tests would fail.
Long answer is: In a real world scenario I always have an availability group with two servers, so I need a shared path for the backups or have to use already available backups.
But in a demo scenario (and in the scenario of the tests) I have an availability group with two instances on the same server. If they run under the same account, the restore can access the local backups. Have a look at the tests: They don't use UseLastBackup nor SharedPath, but the default manual seeding. That works because both instances are on the same server.
So I put these lines there to not forget to start a discussion about that, but commented them out to be able to leave the tests untouched.

Reply:
What tests are you referring to? As well, our commands are not meant to support demo environments and should be written for production situations/configurations; especially for AGs. 

The following code at line 229 is where things will start throwing oddball errors if the user does not provide at least one of those two parameters...that is the whole point of validation and why we do that in our commands (as a standard), to control error messages and ensure they are readable to the user.

```
if ($UseLastBackup) {
                                $allbackups[$db] = Get-DbaDbBackupHistory -SqlInstance $primarydb.Parent -Database $primarydb.Name -IncludeCopyOnly -Last -EnableException
                            } else {
                                $fullbackup = $primarydb | Backup-DbaDatabase -BackupDirectory $SharedPath -Type Full -EnableException
                                $logbackup = $primarydb | Backup-DbaDatabase -BackupDirectory $SharedPath -Type Log -EnableException
                                $allbackups[$db] = $fullbackup, $logbackup
```

If you do not first validate that either `UseLastBackup` parameter or `SharedPath` parameter are provided what output is shown to show the user that a missing parameter was provided to the parent command and that it was just not an issue with the Backup command?

Reply:
As well, if you want to start a discussion on the comment block of code please open an issue and add that block of code...then remove from this PR. If in the end, we decided that it is not required then we do not have to go back and do a PR just to clean up dead code.

Reply:
These tests are failing now:

```
Describe : Add-DbaAgDatabase Integration Tests
Context  : adds ag db
Name     : It returns proper results
Result   : Failed
Message  : Expected 'dbatoolsci_addagdb_agroup', but got $null.

Describe : Get-DbaAgDatabase Integration Tests
Context  : gets ag db
Name     : It returns results
Result   : Failed
Message  : Expected 'dbatoolsci_getagdb_agroup', but got $null.

Describe : New-DbaAvailabilityGroup Integration Tests
Context  : adds an ag
Name     : It returns an ag with a db named
Result   : Failed
Message  : Expected 'dbatoolsci_addag_agroupdb', but got $null.

Describe : Suspend-DbaAgDbDataMovement Integration Tests
Context  : resumes  data movement
Name     : It returns resumed results
Result   : Failed
Message  : Expected 'dbatoolsci_resumeagdb_agroup', but got $null.

```

Reply:
Here is how the test ag is set up:

`$ag = New-DbaAvailabilityGroup -Primary $script:instance3 -Name $agname -ClusterType None -FailoverMode Manual -Database $dbname -Confirm:$false -Certificate dbatoolsci_AGCert
`

So no `SharedPath` and no `UseLastBackup`. And I even had a look at `New-DbaAvailabilityGroup` and the parameter `UseLastBackup`: It is not passed to the `Add-DbaAgDatabase`, so there is another bug.

So I think we have two options: Keep the test and fix `New-DbaAvailabilityGroup` and all the tests or delete the test. As this is a branch and pull request for fixing the 2014 issue, I would like to delete the test here to be able to merge and solve the 2014 issue. The functionality would not be touched, it would just be a fix for 2014. For the parameter `UseLastBackup` we need a general discussion - because it just does not what it says.

@wsmelton What do you think?

Reply:
If the test was passing before and fails from your change, what exactly are we fixing?

Reply:
Foreword: Cooperation across time zones, languages ​​and cultures is not easy and I think we keep talking about each other. 

Please have a look at these lines of code:

```
if ($SeedingMode -eq "Automatic" -and $secondaryInstance.VersionMajor -le 12) {
                    Stop-Function -Continue -Message "Automatic seeding mode not supported on SQL Server versions prior to 2016 - Instance $($secondaryInstance.Name)"

                    if (-not $SharedPath -and -not $UseLastBackup) {
                        Stop-Function -Continue -Message "Automatic seeding not supported and no $SharedPath provided. Database not added to instance $($secondaryInstance.Name)"
                    }
                }
```

There is the test for `$SharedPath` and `$UseLastBackup`. Because of the `Stop-Function`, this code was never executed. So there was never a working test for these parameters. And the pester-tests don't use these parameters and work well. So there are working cases where none of these parameters are needed.
Can we agree on that?

To keep this a non breaking change and only a bugfix for version 2014, I am not allowed to introduce a new test that will let the pester-tests fail. And so I'm also not allowed to change the location of the "inactive" test to a location where it is "active" - because that would be a breaking change.

So I will commit the deletion of the test to fix the bug and not break anything else. 
But I will later on work on the ag commands and especially the parameters `$SharedPath` and `$UseLastBackup` - but consistent in all related commands.

Reply:
`-Continue` this ensures the `Stop-Function` is to continue executing and not throw the full exception.

Reply:
> `-Continue` this ensures the `Stop-Function` is to continue executing and not throw the full exception.

Yes and no. It continues with the command, but does quit the loop. So the `if (-not $SharedPath -and -not $UseLastBackup) {` is never ever executed.

So there never was a test for these parameters. So I can not put "the validation for those two parameters [...] back in the command". I just can't. Sorry to say. I just can't. I can add them to the command - but that would be a breaking change. And I will not do that in a pull request for a bugfix,
And sure I could position them behind a `Stop-Function` like it was - but I will not do that either.